tests/model/model.cpp

```diff
@@ -624,3 +624,491 @@ TEST(Model, removeComponentInsensitiveToOrder)
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+
+    EXPECT_TRUE(model->removeUnits("myConcreteUnits"));
+    EXPECT_TRUE(model->removeUnits("myImportedUnits"));
+
+    EXPECT_EQ(size_t(0), model->unitsCount());
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+}
+
+TEST(Model, removeImportedUnitsByIndex)
+{
+    auto model = libcellml::Model::create();
+    auto myConcreteUnits = libcellml::Units::create("myConcreteUnits");
+    auto myImportedUnits = libcellml::Units::create("myImportedUnits");
+
+    auto import = libcellml::ImportSource::create();
+    import->setUrl("import.cellml");
+    myImportedUnits->setImportSource(import);
+
+    model->addUnits(myConcreteUnits);
+    model->addUnits(myImportedUnits);
+
+    EXPECT_TRUE(model->units("myImportedUnits")->isImport());
+    EXPECT_FALSE(model->units("myConcreteUnits")->isImport());
+
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+
+    EXPECT_TRUE(model->removeUnits(0));
+    EXPECT_TRUE(model->removeUnits(0));
+
+    EXPECT_EQ(size_t(0), model->unitsCount());
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+}
+
+TEST(Model, removeImportedUnitsByReference)
+{
+    auto model = libcellml::Model::create();
+    auto myConcreteUnits = libcellml::Units::create("myConcreteUnits");
+    auto myImportedUnits = libcellml::Units::create("myImportedUnits");
+
+    auto import = libcellml::ImportSource::create();
+    import->setUrl("import.cellml");
+    myImportedUnits->setImportSource(import);
+
+    model->addUnits(myConcreteUnits);
+    model->addUnits(myImportedUnits);
+
+    EXPECT_TRUE(model->units("myImportedUnits")->isImport());
+    EXPECT_FALSE(model->units("myConcreteUnits")->isImport());
+
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+
+    EXPECT_TRUE(model->removeUnits(myConcreteUnits));
+    EXPECT_TRUE(model->removeUnits(myImportedUnits));
+
+    EXPECT_EQ(size_t(0), model->unitsCount());
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+}
+
+TEST(Model, takeImportedUnitsByName)
+{
+    auto model = libcellml::Model::create();
+    auto myConcreteUnits = libcellml::Units::create("myConcreteUnits");
+    auto myImportedUnits = libcellml::Units::create("myImportedUnits");
+
+    auto import = libcellml::ImportSource::create();
+    import->setUrl("import.cellml");
+    myImportedUnits->setImportSource(import);
+
+    model->addUnits(myConcreteUnits);
+    model->addUnits(myImportedUnits);
+
+    EXPECT_TRUE(model->units("myImportedUnits")->isImport());
+    EXPECT_FALSE(model->units("myConcreteUnits")->isImport());
+
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+
+    auto takeUnits1 = model->takeUnits("myImportedUnits");
+    auto takeUnits2 = model->takeUnits("myConcreteUnits");
+
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+    EXPECT_EQ(size_t(0), model->unitsCount());
```

Comment:
Might want to swap the two tests? Just so that it's "consistent" with what has been done in the above tests (makes it easier to review, maintain, etc.) (Same below.)

Reply:
e7a8784